apps/base-docs/docusaurus.config.js

```diff
@@ -1,3 +1,7 @@
+const dotenv = require('dotenv');
+dotenv.config();
+dotenv.config({ path: '.env.local', override: true });
```

Comment:
Think U probably don't want to do this

Reply:
Hm, unfortunately this will prove difficult due to the way the Dockerfile build works. We'll only have access to environment secrets at _server_ runtime but not at build time. The whole workaround with NextJS publicRuntimeConfig is to get around this limitation. I don't have a clean solution for Docusaurus off the top of my head sadly.

Reply:
I was trying to get around this by using `customFields` in the Docusaurus config, like they recommend [here](https://docusaurus.io/docs/2.0.1/deployment#:~:text=in%20a%20typical,the%20client%20side.). Then in my init code [here](https://github.com/base-org/web/pull/29/files#diff-a2a9430c92253339293f1d13caa312b572720fd7c9897ec6b43e89bd8adf6606R5-R23), I'm using env variable values via the customFields pulled from the generated config file.

I don't have all the context to understand the limitation we run into when deploying to prod, but will this still not work?

Reply:
Yeah, I think this probably still won't work. There's a problem with the difference between ENV vars available at _build_ time vs ENV vars available at _run_ time. Docusaurus is really oriented around doing everything at build time which makes it tough for us because we inject secret vars at run time.

Reply:
Do you think my best path forward [here](https://github.com/base-org/web/pull/29/files#diff-a2a9430c92253339293f1d13caa312b572720fd7c9897ec6b43e89bd8adf6606R16-R20) is to just set isProd to true, showDebugLogging to false, and paste the API key into the code? That's what was done in [this onchainsummer example](https://github.com/base-org/onchainsummer.xyz/blob/fb437f2716843006f741a3ae94a816c7f640da45/public/init_cca.js#L5), but I really wanted to avoid it since dev data will end up getting sent to prod amplitude.

Reply:
Ok, I think we can make this work. Basically hardcode things so that `production` works by default, and then override things on development so it doesn't send stats. This works because in development we have access to ENV vars at both build and runtime so overriding in development would work.

The onchainsummer example would basically be the hardcoded base case that U'll override with local development ENV.

Reply:
Sounds like a good workaround. How would I accomplish the override in this case? Are you suggesting I do something like this for lines in the init code that use env vars: `amplitudeApiKey: process.env.amplitudeApiKey || 'prod-api-key'`?

Reply:
I'm not exactly sure what form the code will take but I think if U experiment with it given the spec 'works by default like prod, overridden in development' I have full confidence U'll figure it out!

Reply:
My bad, I'd assumed you had something specific in mind. Sounds good though! I'll take a pass at that now.

Reply:
All good! Thanks for bearing through the difficulties of working in this particular environment :)

Reply:
Will using process.env.NODE_ENV cause errors in docs prod or just come up as undefined? If it doesn't cause any errors, I think I can just remove the dotenv implementation. With that removed, I could go back to referencing process.env in the docs analytics init code as long as I provide values for prod to fallback on when they're undefined. So it'd be like this:
- const isDevelopment = process.env.NODE_ENV === 'development'
- isProd: !isDevelopment
- showDebugLogging: isDevelopment
- amplitudeApiKey: isDevelopment ? 'development-key' : 'prod-key'

Reply:
Will using env variables via docusaurus customFields cause errors in prod or just be undefined? Assuming they're just undefined, I have a solution that I can commit in the morning. Otherwise, I'll take another pass at this then.